tools/run_tests/run_tests.py

```diff
@@ -308,6 +308,13 @@ def configure(self, config, args):
 
     def test_specs(self):
         out = []
+        if self.platform == 'mac' and self.test_lang == 'c++':
```

Comment:
this feels like a hack. we should come up with a better place for these tests rather than gluing on run_tests.py logic for C/C++ which is already complex enough. Also semantically, the tests don't belong here.

Reply:
Sure. I think I'll put all the iOS tests in one separate job from the MacOS tests so we pay the download overhead only once.